legal-api/src/legal_api/services/authz.py

```diff
@@ -95,96 +107,441 @@ def has_roles(jwt: JwtManager, roles: List[str]) -> bool:
 ALLOWABLE_FILINGS: Final = {
     'staff': {
         Business.State.ACTIVE: {
-            'adminFreeze': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
-            'alteration': ['BC', 'BEN', 'ULC', 'CC'],
-            'annualReport': ['CP', 'BEN', 'BC', 'ULC', 'CC'],
-            'changeOfAddress': ['CP', 'BEN', 'BC', 'ULC', 'CC'],
-            'changeOfDirectors': ['CP', 'BEN', 'BC', 'ULC', 'CC'],
-            'changeOfRegistration': ['SP', 'GP'],
-            'consentContinuationOut': ['BC', 'BEN', 'CC', 'ULC'],
-            'conversion': ['SP', 'GP', 'BEN'],
-            'correction': ['CP', 'BEN', 'SP', 'GP', 'BC', 'ULC', 'CC'],
-            'courtOrder': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
-            'dissolution': ['CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC', 'SP', 'GP'],
-            'incorporationApplication': ['CP', 'BC', 'BEN', 'ULC', 'CC'],
-            'registrarsNotation': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
-            'registrarsOrder': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
-            'registration': ['SP', 'GP'],
-            'specialResolution': ['CP'],
-            'transition': ['BC', 'BEN', 'CC', 'ULC'],
+            'adminFreeze': {
+                'legalTypes': ['SP', 'GP', 'CP', 'BC', 'BEN', 'CC', 'ULC', 'LLC'],
+            },
+            'alteration': {
+                'legalTypes': ['BC', 'BEN', 'ULC', 'CC'],
+                'blockerChecks': {
+                    'business': [BusinessBlocker.DEFAULT]
+                }
+            },
+            'annualReport': {
+                'legalTypes': ['CP', 'BEN', 'BC', 'ULC', 'CC'],
+                'blockerChecks': {
+                    'business': [BusinessBlocker.DEFAULT]
+                }
+            },
+            'changeOfAddress': {
+                'legalTypes': ['CP', 'BEN', 'BC', 'ULC', 'CC'],
+                'blockerChecks': {
+                    'business': [BusinessBlocker.DEFAULT]
+                }
+            },
+            'changeOfDirectors': {
+                'legalTypes': ['CP', 'BEN', 'BC', 'ULC', 'CC'],
+                'blockerChecks': {
+                    'business': [BusinessBlocker.DEFAULT]
+                }
+            },
+            'changeOfRegistration': {
+                'legalTypes': ['SP', 'GP'],
+                'blockerChecks': {
+                    'warningTypes': [WarningType.MISSING_REQUIRED_BUSINESS_INFO],
+                    'business': [BusinessBlocker.DEFAULT]
+                }
```

Comment:
`warningTypes ` attribute is used exclude a filing if a given `warningType` exists for given business.

As additional warnings are added,  `warningTypes` can be used where applicable.